api/progress.go

```diff
@@ -114,10 +116,37 @@ func (r progressRoutes) saveProgress(c *gin.Context) {
 		})
 		return
 	}
+
+	streamIDString := strconv.FormatUint(uint64(request.StreamID), 10)
+	if r.StreamsDao == nil {
+		logger.Error("StreamsDao is nil")
+		return
+	}
+	stream, err := r.StreamsDao.GetStreamByID(c, streamIDString)
+
+	if err == nil {
+		lastSilence := slices.MaxFunc(stream.Silences, func(silence model.Silence, other model.Silence) int {
+			return int(silence.End) - int(other.End)
+		})
+
+		if math.Abs(float64(lastSilence.End-uint(stream.Duration.Int32))) < 10 {
+			lastSilencePercent := float64(lastSilence.Start) / float64(stream.Duration.Int32)
+			if request.Progress >= lastSilencePercent {
+				progressBuff.add(model.StreamProgress{
+					Progress: request.Progress,
+					StreamID: request.StreamID,
+					UserID:   tumLiveContext.User.ID,
+					Watched:  true,
+				})
+				return
```

Comment:
rather don't return but, assign to a var `watchedToLastSilence` or something here and write to buffer `Watched: request.Progress > .9 || watchedToLastSilence` later to avoid duplication